hawtjni-generator/src/main/java/org/fusesource/hawtjni/generator/model/ReflectField.java

```diff
@@ -100,6 +100,14 @@ public boolean isPointer() {
         return getFlag(POINTER_FIELD) || ( type.getWrappedClass() == Long.TYPE && getCast().endsWith("*") );
     }
 
+    public boolean isSharedPointer() {
+        if( annotation == null ) {
+
+             return false;
+        }
+        return getFlag(SHARED_PTR) || ( type.getWrappedClass() == Long.TYPE && getCast().endsWith("*") );
```

Comment:
Are you sure about the `||` clause ? This means that any type which is a `long` and cast to a pointer will use a shared pointer ?  I think this should be an opt-in with the `SHARED_PTR` flag imho, so that we can keep compatibility.  So I think it should just be
```
return getFlag(SHARED_PTR);
```

Reply:
yes || clause is not required even return getFlag(SHARED_PTR); is sufficient ..